bubblesub/ui/views.py

```diff
@@ -0,0 +1,108 @@
+# bubblesub - ASS subtitle editor
+# Copyright (C) 2018 Marcin Kurczewski
+#
+# This program is free software: you can redistribute it and/or modify
+# it under the terms of the GNU General Public License as published by
+# the Free Software Foundation, either version 3 of the License, or
+# (at your option) any later version.
+#
+# This program is distributed in the hope that it will be useful,
+# but WITHOUT ANY WARRANTY; without even the implied warranty of
+# MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+# GNU General Public License for more details.
+#
+# You should have received a copy of the GNU General Public License
+# along with this program.  If not, see <https://www.gnu.org/licenses/>.
+
+from PyQt5 import QtCore, QtWidgets
+
+from bubblesub.api import Api
+from bubblesub.api.gui import ViewLayout
+from bubblesub.ui.audio import Audio
+from bubblesub.ui.video import Video
+
+
+class ViewLayoutManager(QtCore.QObject):
+    def __init__(
+        self,
+        api: Api,
+        video: Video,
+        audio: Audio,
+        main_window: QtWidgets.QMainWindow,
+    ) -> None:
+        super().__init__(main_window)
+
+        self._api = api
+        self._video = video
+        self._audio = audio
+        self._main_window = main_window
+        self._view_layout = ViewLayout.Full
+
+    @property
+    def view_layout(self) -> ViewLayout:
+        return self._view_layout
+
+    @view_layout.setter
+    def set_view_layout(self, view_layout: ViewLayout) -> None:
+        self._run_view(view_layout)
```

Comment:
This shouldn't be a setter, not with such name. Or you can change this name to `view_layout` and leave it as a setter